src/middlewared/middlewared/plugins/account.py

```diff
@@ -570,22 +570,24 @@ def shell_choices(self):
         if password:
             data['unixhash'] = crypted_password(password)
             # See http://samba.org.ru/samba/docs/man/manpages/smbpasswd.5.html
-            data['smbhash'] = f'{data["username"]}:{data["uid"]}:{"X" * 32}:{nt_password(password)}:[U          ]:LCT-{int(time.time()):X}:'
+            if data['locked']:
+                data['smbhash'] = f'{data["username"]}:{data["uid"]}:{"X" * 32}:{nt_password(password)}:[DU         ]:LCT-{int(time.time()):X}:'
+            else:
+                data['smbhash'] = f'{data["username"]}:{data["uid"]}:{"X" * 32}:{nt_password(password)}:[U          ]:LCT-{int(time.time()):X}:'
         else:
             data['unixhash'] = '*'
             data['smbhash'] = '*'
         return password
 
-    async def __set_smbpasswd(self, username, password):
+    def __set_smbpasswd(self, username):
```

Comment:
Why is this method synchronous? It does not do anything synchronous and it is always called from asynchronous context.

Reply:
Fixed.